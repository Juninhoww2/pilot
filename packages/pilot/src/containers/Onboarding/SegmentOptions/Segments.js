import React from 'react'
import PropTypes from 'prop-types'
import { Button } from 'former-kit'
import Card from '../Card'
import ArrowBack from '../arrow-back.svg'
import styles from './styles.css'

const Segments = ({
  handleReturn,
  handleSubmit,
  options,
  subtitle,
}) => (
  <div>
    <div className={styles.segmentReturn}>
      <Button
        icon={<ArrowBack width={16} height={16} />}
        fill="clean"
        onClick={handleReturn}
      />
      <span>{subtitle}</span>
    </div>
    <div className={styles.segmentCardsWrapper}>
      {options.map(option => (
        <Card
          key={option.value}
          onSubmit={() => handleSubmit(option.value)}
        >
          <p className={styles.segmentCardContent}>
            {option.label}
          </p>
        </Card>
      ))}
    </div>
  </div>
)

Segments.propTypes = {
  handleReturn: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.object).isRequired,
  subtitle: PropTypes.string.isRequired,
}

export default Segments
