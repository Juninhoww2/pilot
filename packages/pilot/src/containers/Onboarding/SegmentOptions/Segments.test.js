import React from 'react'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Segments from './Segments'

test('Segments should render', () => {
  const subtitle = 'Turismo e eventos'
  const options = [
    {
      label: 'Ingressos',
      value: 'ticket',
    }, {
      label: 'Desconto coletivo',
      value: 'discount',
    },
  ]
  const handleSubmit = jest.fn()
  const handleReturn = jest.fn()

  const {
    container, getByText,
  } = render(
    <Segments
      handleReturn={handleReturn}
      handleSubmit={handleSubmit}
      options={options}
      subtitle={subtitle}
    />
  )
  userEvent.click(container.querySelector('button'))
  expect(handleReturn).toHaveBeenCalledTimes(1)

  userEvent.click(getByText(options[0].label))
  expect(handleSubmit).toHaveBeenCalledTimes(1)

  expect(container).toMatchSnapshot()
})
