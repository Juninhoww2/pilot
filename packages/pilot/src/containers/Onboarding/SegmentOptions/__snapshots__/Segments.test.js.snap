// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Segments should render 1`] = `
<div>
  <div>
    <div
      class="segmentReturn"
    >
      <button
        class=""
        type="button"
      >
        <svg
          data-file-name="SvgArrowBack"
          height="16"
          width="16"
        />
        <span
          style="height: 0px; left: 0px; top: 0px; width: 0px;"
        />
      </button>
      <span>
        Turismo e eventos
      </span>
    </div>
    <div
      class="segmentCardsWrapper"
    >
      <div
        class="clickableCard"
        role="button"
        tabindex="0"
      >
        <div
          class="card"
        >
          <p
            class="segmentCardContent"
          >
            Ingressos
          </p>
        </div>
      </div>
      <div
        class="clickableCard"
        role="button"
        tabindex="0"
      >
        <div
          class="card"
        >
          <p
            class="segmentCardContent"
          >
            Desconto coletivo
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
